internal/sizes/predict.go

```diff
@@ -0,0 +1,135 @@
+package sizes
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/ViaQ/logerr/kverrors"
+
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+)
+
+var (
+	metricsClient Client
+
+	// durationPredictSecs is the value of time series in seconds from now.
+	// It is passed as second parameter to predict_linear.
+	durationPredictSecs = 1 * 24 * 3600 // 1d = 1 * 24h * 3600s = 86400s
+	// timeoutClient is the timeout duration for prometheus client.
+	timeoutClient = 10 * time.Second
+
+	// promURL is the URL of the prometheus thanos querier
+	promURL string
+	// promToken is the token to connect to prometheus thanos querier.
+	promToken string
+
+	cmd []byte
+)
+
+type client struct {
+	api     v1.API
+	timeout time.Duration
+}
+
+// Client is the interface which contains methods for querying and extracting metrics.
+type Client interface {
+	LogLoggedBytesReceivedTotal(duration model.Duration) (float64, error)
+}
+
+func newClient(url, token string) (*client, error) {
+	httpConfig := config.HTTPClientConfig{
+		BearerToken: config.Secret(token),
+		TLSConfig: config.TLSConfig{
+			InsecureSkipVerify: true,
```

Comment:
We can omit this on OpenShift by mounting the [Service CA](https://docs.openshift.com/container-platform/4.8/security/certificate_types_descriptions/service-ca-certificates.html) via a ConfigMap annotated with `service.beta.openshift.io/inject-cabundle: true`